### PR TITLE
Correcting bugs with the WFMarket AFK system

### LINKED ISSUE #135: WFM status locking
**Describe the bug**
Wfm status is not changed from the initial value if the user does not manually update it himself first.

**To Reproduce**
1. Log into WFM using "remember me"
2. Restart application without warframe running
3. AFK for >7 minutes
4. See status not being changed off from "online"

**Expected behavior**
Status being dynamic and changing when needed

### LINKED ISSUE #222: Restore last known user-status when returning from AFK instead of just "online"
**Is your feature request related to a problem? Please describe.**
I've had multiple times where I've either left Warframe running and set my status to offline, played for a while then away for a moment to do something only to be surprised with market messages sometime later after coming back.

The app always sets you to online on returning even if that is not what you want or if the game isn't even open.

**Describe the solution you'd like**
On returning the app restores the status it last saw you as before going AFK.

**Additional context**
This is also probably related to problems seen in issue #135 

See related pull request for suggested fix

### PR DESCRIPTION
**Pull request by PeterJeff**
- Fixed a logical error where while AFK the system would alternate between offline and online or setting online when AFK and Warframe was closed
- Fixed multiple issues of AFK system sending status change requests to WFMarket when no change was needed or the status was already set
- Added status tracking for AFK to restore the last known user-set status instead of simply setting to online in-game or online which may be undesirable

### What did you fix? 
closed #135 WFM status locking
closes #222 Restore last known user-status when returning from AFK instead of just "online"

---

### Reproduction steps
1. Run Warframe & WFinfo when logged into warframe.market in the app.  
2. Confirm online in-game status
3. Do not press any keys or click for 7-8 minutes.  Leave website and app status visible
4. Confirm that the app has set the user to invisible/offline status
5. Press buttons or keys while waiting up to 1 minute
6. Confirm that the app has set the user to online in-game on WFMarket

Try setting other WFMarket statuses before going AFK and confirm they are restored on returning. 

Time can be compressed by replacing code that uses `minutesTillAfk` to AddSeconds instead of AddMinutes and changing periodTimeSpan to some number of seconds instead of 1 minute.

---

### Evidence/screenshot/link to line

You can see my fixes highlighted in comments below

### Considerations
- Does this contain a new dependency? **No**
- Does this introduce opinionated data formatting or manual data entry? **No**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **No**
- Have I run the linter? **No**
- Is is a bug fix, feature request, or enhancement? **Bug Fix & Enhancement**

### CODE REVIEW

File: WFInfo/Main.cs [hunk 1]
```diff
@@ -101,40 +103,58 @@ public static void ThreadedDataLoad()
...
         {
-            if (!await dataBase.IsJWTvalid())
+            if (!await dataBase.IsJWTvalid().ConfigureAwait(true))
                 return;
-            var now = DateTime.UtcNow;
+            DateTime now = DateTime.UtcNow;
             Debug.WriteLine($"Checking if the user has been inactive \nNow: {now}, Lastactive: {latestActive}");
 
-            if (OCR.Warframe != null && OCR.Warframe.HasExited)
+            if (OCR.Warframe != null && OCR.Warframe.HasExited && LastMarketStatus != "invisible")
```
Comment (PeterJeff) [inline WFInfo/Main.cs:111]: added status awareness

File: WFInfo/Main.cs [hunk 2]
```diff
@@ -101,40 +103,58 @@ public static void ThreadedDataLoad()
...
-                    await dataBase.SetWebsocketStatus("invisible");
+                    //IDE0058 - computed value is never used.  Ever. Consider changing the return signature of SetWebsocketStatus to void instead
+                    await dataBase.SetWebsocketStatus("invisible").ConfigureAwait(false);
                     StatusUpdate("WFM status set offline, Warframe was closed", 0);
-                });
+                }).ConfigureAwait(false);
             }
-            
-            if (latestActive <= now)
+            else if (UserAway && latestActive > now)
```
Comment (PeterJeff) [inline WFInfo/Main.cs:124]: conditions are now else instead of all being simultaneously possible (which is nonsense)

File: WFInfo/Main.cs [hunk 3]
```diff
@@ -412,6 +432,13 @@ public static void FinishedLoading()
         }
         public static void UpdateMarketStatus(string msg)
         {
+            Debug.WriteLine($"New market status received: {msg}");
+            if (!UserAway)
```
Comment (PeterJeff) [inline WFInfo/Main.cs:436]: Status is only stored if the user is not away.  automated AFK status changes are irrelevant.

File: WFInfo/Data.cs [hunk 4]
```diff
@@ -1281,6 +1281,7 @@ public async Task<bool> SetWebsocketStatus(string status)
             var message = "{\"type\":\"@WS/USER/SET_STATUS\",\"payload\":\"";
             switch (status)
             {
+                case "ingame":
```
Comment (PeterJeff) [inline WFInfo/Data.cs:1284]: This bothers me but I don't know what to do about it

**Comment by dapal-003** [comment]
Hey thanks for the pr, I've ran into two issues with testing this.
1: The timer is not initialized if the user logs in for the first time
The timer is currently after a login check that is ran at launch, so the timer doesn't start if the user logs in. An easy fix is to put the timer behind the LoggedIn method https://github.com/WFCD/WFinfo/blob/c2d18e4f6a2b34a65df49dee296be8f826b14817/WFInfo/Main.cs#L403

2: The UI doesn't update when the timer puts the user offline, I currently don't know why the status doesn't get updated but the drop down doesn't have any code to change it

**Comment by PeterJeff** [comment]
> Hey thanks for the pr, I've ran into two issues with testing this. 1: The timer is not initialized if the user logs in for the first time The timer is currently after a login check that is ran at launch, so the timer doesn't start if the user logs in. An easy fix is to put the timer behind the LoggedIn method
> 
> https://github.com/WFCD/WFinfo/blob/c2d18e4f6a2b34a65df49dee296be8f826b14817/WFInfo/Main.cs#L403
> 
> 2: The UI doesn't update when the timer puts the user offline, I currently don't know why the status doesn't get updated but the drop down doesn't have any code to change it

I'll look at them and fix it

**Comment by PeterJeff** [comment]
> 2: The UI doesn't update when the timer puts the user offline, I currently don't know why the status doesn't get updated but the drop down doesn't have any code to change it

I spent more time being confused by this than I'd like to admit before I realized it's bug on master.

The last commit where the warframe.market status feature works (can build and run without crashing) is this:
https://github.com/WFCD/WFinfo/commit/cc7d7b0a3c4f0dc72f78c069e322117d21490314

**Comment by D1firehail** [comment]
Is there anything holding this up other than waiting for your other PR (#230, although at the time of writing that one has a small issue)? The commit after the one you mentioned appears to be when the incorrect logic in `IsJwtAvailable` was introduced, which is likely what broke this one

**Comment by dapal-003** [comment]
> Is there anything holding this up
There is one small conflict that must be resolved. Not sure if I have the perms to resolve that merge contlift though

**Comment by dapal-003** [comment]
I was able to do so

**Pull request merged**